src/app/app.component.ts

```diff
@@ -40,54 +44,13 @@ export class AppComponent extends withUnsubscribe() implements OnInit {
       withLatestFrom(this.loader.value$),
       map((v) => v[1])
     );
-
-    // Update page layout after every navigation change
-    this.router.events
-      .pipe(
-        filter((event) => event instanceof NavigationEnd),
-        takeUntil(this.unsubscribe)
-      )
-      .subscribe(() => this.updatePageLayout(), noop);
   }
 
   /**
    * Determine whether the currently shown component uses the menu layout or fullscreen
    */
-  private updatePageLayout() {
-    // Default to fullscreen
-    this.fullscreen = true;
-
-    // Find the primary router component
-    let displayComponent = this.route.firstChild;
-    let count = 0;
-
-    // Continue searching to a depth of 50 components (prevent circular looping)
-    while (count < 50) {
-      if (!displayComponent) {
-        // This is not a display component, cannot go any further
-        console.error("Could not find a display component.");
-        return;
-      } else if (displayComponent.component) {
-        // This is a display component, break out of while loop and check page info
-        break;
-      }
-
-      // Component has a child, keep digging further down
-      displayComponent = displayComponent.firstChild;
-      count++;
-    }
-
-    if (count >= 50) {
-      console.error("Search for component layout type exceeded a depth of 50.");
-      return;
-    }
-
-    // Check if component is a page info component and is not set to fullscreen
-    const component: PageComponent = displayComponent.component as any;
-    if (!component.pageInfo) {
-      this.fullscreen = true;
-    } else {
-      this.fullscreen = !!component.pageInfo.fullscreen;
-    }
+  public updatePageLayout(component: any) {
+    this.fullscreen = !!(component?.constructor?.pageInfo as PageInfo)
+      ?.fullscreen;
```

Comment:
Since the mechanics of this mostly happen in templates, add a comment explaining workflow...

When component is loaded, uses its knowledge of pageInfo, to call this method (to escape encapsualtion) and change the fullscreen property, which is bound ??by/from/to what??